### DISCUSSION TITLE
How to use memory to answer the next question if the previous question's context matches with the current one.

### DISCUSSION THREAD

**Discussion by nupurkhare-15** [crewAIInc/crewAI#1099] [Q&A]
When asking about a certain topic for example, A and then again asking - Where does he live? 
Then its not generating correct answer.
Ideally it should have taken previous answer into context and then answer the next question.
How should memory usage be taken into consideration.

**Comment by theCyberTech** [comment]
Sorry but your question is hard to understand can you show us what your crew looks like as in Agents , tasks on what process

**Reply by TobiasMaissen** [reply]
I have (had) the same Issue. Long Story Short: if you set the memory=True within the crew it seems that it won't make any difference. No context available.
I've done a Setup where an Agent queries a PostgreSQL DB. Which works absolutely fine.
Example:
Me: Which Customer makes the most revenue? --> Agent Result: Customer XY with XY USD
Me: Okay and where does he lives? --> Agent: Here is a list of all customers

After researching a lot I found a solution which not at any point revers to the docs from crewAI:
To really achiev the memory goal with context -->
1. Must set the Variable CREWAI_STORAGE_DIR
2. having the following code.

First, why is it so hidden to reach the crewAI Memory? I thougt this would work out of the box..
Second, is it possible to adapt the docs more specific to that. Or is it an Issue with the Storage Dir Variable?

`
# Initialize the Long Term Memory
ltm = LongTermMemory(
    storage=LTMSQLiteStorage(
        db_path=f"{DATA_DIR}/long_term_memory.db")
)

# Initialize the Short Term Memory
stm = ShortTermMemory(
    storage=RAGStorage(
        type="short_term",
        embedder_config=embedder
    )
)

# Initialize the Entity Memory
em = EntityMemory(
    storage=RAGStorage(
        type="entities",
        embedder_config=embedder
    )
)

    @crew
    def crew(self) -> Crew:
        return Crew(
            agents=self.agents,
            tasks=self.tasks,
            process=Process.sequential,
            verbose=True,
            memory=True,
            long_term_memory=ltm,
            short_term_memory=stm,
            entity_memory=em
        )
`
